### PR TITLE
threaded writing for BAM files

### PR DESCRIPTION
**Pull request by pmarks**
activate htslib support for multi-threaded BAM writing.

### CODE REVIEW

File: src/bam/mod.rs [hunk 1]
```diff
@@ -805,6 +819,56 @@ mod tests {
...
+        println!("{:?}", bampath);
+        {
+            let mut bam = Writer::from_path(
+                &bampath,
+                Header::new().push_record(
+                    HeaderRecord::new(b"SQ").push_tag(b"SN", &"chr1")
+                                            .push_tag(b"LN", &15072423)
+                )
+            ).ok().expect("Error opening file.");
+            bam.set_threads(4).ok();
```
Comment (johanneskoester) [inline src/bam/mod.rs]: Can you also add an expect?

File: src/bam/mod.rs [hunk 2]
```diff
@@ -358,6 +358,20 @@ impl Writer {
         Ok(Writer { f: f, header: HeaderView::new(header_record) })
     }
 
+    /// Activate multi-threaded BAM write support in htslib. This should permit faster
+    /// writing of large BAM files.
+    /// # Arguments
+    ///
+    /// * `n_threads` - number of background writer threads to use
+    pub fn set_threads(&mut self, n_threads: usize) -> Result<(), WriteError> {
```
Comment (johanneskoester) [inline src/bam/mod.rs]: Maybe have a separate error type for this? ThreadingError::Some? With a description like "error setting threads for multithreaded writing".

File: src/bam/mod.rs (hunk 1 above)
Comment (johanneskoester) [inline src/bam/mod.rs]: If we have the error type I outline above, this can simply be an unwrap call.

**Comment by pmarks** [comment]
@johanneskoester - sorry, forgot about this one for a while.

**Comment by johanneskoester** [comment]
No problem, thank you!

**Pull request merged**